### PR TITLE
[Feature] Add cutmix option

### PR DESCRIPTION
**Pull request by HIT-cwh**
Hi,
This pr implements cutmix.
Pls kindly take a look.
Thank you.

### CODE REVIEW

**Review by MeowZheng — COMMENTED** [review]
please check img and gt after cutmix ref [mixup](https://github.com/open-mmlab/mmclassification/pull/160/files)

**Review by MeowZheng — APPROVED** [review]

**Comment by ycxioooong** [comment]
How about the benchmark regression result of cutmix?

File: configs/_base_/models/resnet50_cifar_cutmix.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        num_stages=4,
+        out_indices=(3, ),
+        style='pytorch'),
+    neck=dict(type='GlobalAveragePooling'),
+    head=dict(
+        type='MultiLabelLinearClsHead',
+        num_classes=10,
+        in_channels=2048,
+        loss=dict(type='CrossEntropyLoss', loss_weight=1.0, use_soft=True)),
+    train_cfg=dict(cutmix=dict(alpha=1.0, num_classes=10)))
```
Comment (d-li14) [inline configs/_base_/models/resnet50_cifar_cutmix.py:16]: @HIT-cwh The `cutmix_prob` seems to be missing here.

File: configs/_base_/models/resnet50_cutmix.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        num_stages=4,
+        out_indices=(3, ),
+        style='pytorch'),
+    neck=dict(type='GlobalAveragePooling'),
+    head=dict(
+        type='MultiLabelLinearClsHead',
+        num_classes=1000,
+        in_channels=2048,
+        loss=dict(type='CrossEntropyLoss', loss_weight=1.0, use_soft=True)),
+    train_cfg=dict(cutmix=dict(alpha=1.0, num_classes=1000)))
```
Comment (d-li14) [inline configs/_base_/models/resnet50_cutmix.py:16]: The same as above.

**Review by d-li14 — COMMENTED** [review]
Could you pls take a look at this? Thanks.

File: configs/_base_/models/resnet50_cifar_cutmix.py (hunk 1 above)
Comment (HIT-cwh) [reply configs/_base_/models/resnet50_cifar_cutmix.py:16]: Great point. Will fix ASAP

**Pull request merged**